library/src/data/index.js

```diff
@@ -41,20 +41,265 @@ window.CoNDeT.data = {
+      var editTable = getTableWithId(newState, tableId);
+      editTable.rows.push(newRow);
+
+      this.stateManager.setState(newState);
+    };
+
+    removeRow = function (tableId, rowId) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (editTable.rows != null) {
+        for (var i = 0; i < editTable.rows.length; i++) {
+          if (editTable.rows[i].id != rowId) continue;
+          editTable.rows.splice(rowId, 1);
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    addConditionColumn = function (tableId, column) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (column != null) {
+        editTable.columns.conditions.push(column);
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    addDecisionColumn = function (tableId, column) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (column != null) {
+        editTable.columns.decisions.push(column);
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    removeConditionColumn = function (tableId, column) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      removeColumnWithId(
+        editTable.columns.conditions,
+        editTable.rows,
+        "conditions",
+        column
+      );
+
+      this.stateManager.setState(newState);
+    };
+
+    removeDecisionColumn = function (tableId, column) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      removeColumnWithId(
+        editTable.columns.decisions,
+        editTable.rows,
+        "decisions",
+        column.decision
+      );
+
+      this.stateManager.setState(newState);
+    };
+
+    changeOrder = function (tableId) {}; //WIP
+
+    addConnection = function (tableId, rowId, secondTableId) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      for (var i = 0; i < editTable.rows.length; i++) {
+        if (editTable.rows[i].id == rowId) {
+          editTable.rows[i].connections.push(secondTableId);
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    removeConnection = function (tableId, rowId, secondTableId) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      for (var i = 0; i < editTable.rows.length; i++) {
+        if (editTable.rows[i].id !== rowId) continue;
+        for (var j = 0; j < editTable.rows[i].connections.length; j++) {
+          if (editTable.rows[i].connections[j] !== secondTableId) connections;
+          editTable.rows[i].connections.splice(j, 1);
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    replaceInArray = function (toReplace, newValues) {
+      if (newValues != []) {
```

Comment:
Czy to nie zwróci zawsze `true` ?
No bo `[]` to object, czyli masz porównanie by reference. A przynajmniej kiedy by było `!==`
Sprawa się komplikuje przy `!=` bo masz coercion. Więc jak podasz typ prosty to zaczynają się cyrki. Np.:
* `"" == []` -> `true`
* `0 == []` -> `true`
* `[] == []` -> `false`
* `[] == false` -> `true` // jeżeli dobrze rozumuję to `[].toString()` da `""` co jest falsy value więc będzie `true`

Także według mnie `==` i `!=` powinno się używać tylko z `null` i `undefined` (bo one są porównywane tylko z sobą, dalsze coercion nie następuje). Generalnie używanie `===` pozwala uniknąć większości tych cyrków i zaskoczeń. A jak już chcemy korzystać z `==` to polecam mieć [dokumentację](https://tc39.es/ecma262/#sec-abstract-equality-comparison) w jednym palcu.
A kończąc wypowiedź, to uważam, że ten if powinien zostać zniszczony